### PR TITLE
Disable/Enable rate limit for erc20 tokens 

### PR DESCRIPTION
**Pull request by viraj124**
- closes https://github.com/FuelLabs/fuel-bridge/issues/309

### CODE REVIEW

File: packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol [hunk 1]
```diff
@@ -89,6 +89,11 @@ contract FuelERC20GatewayV4 is
     /// @notice The eth withdrawal limit amount for each token.
     mapping(address => uint256) public limitAmount;
 
+    /// @notice Flag to indicate rate limit status for each token, whether disabled or enabled.
+    // if `true` it is disabled
+    // if `false` it is enabled
```
Comment (DefiCake) [inline packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol:94]:
```suggestion
    // if `false` it is disabled
    // if `true` it is enabled
```

File: packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol [hunk 2]
```diff
@@ -175,6 +180,15 @@ contract FuelERC20GatewayV4 is
...
         emit RateLimitUpdated(_token, _amount);
     }
+    
+    /**
+     * @notice updates rate limit status by disabling/re-enabling rate limit.
+     * @param _token The token address to update rate limit status for.
+     * @param _rateLimitStatus bool flag to disable or re-enable rate limit.
+     */
+    function updateRateLimitStatus(address _token, bool _rateLimitStatus) external onlyRole(SET_RATE_LIMITER_ROLE) {
+        rateLimitStatus[_token] = _rateLimitStatus;
```
Comment (DefiCake) [inline packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol:190]: Add an event please

**Review by DefiCake — APPROVED** [review]

File: packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol [hunk 3]
```diff
@@ -298,6 +318,12 @@ contract FuelERC20GatewayV4 is
         emit Withdrawal(bytes32(uint256(uint160(to))), tokenAddress, amount);
     }
 
+    /// @notice Checks if the rate limit is enabled or not
+    /// @param tokenAddress Address of the token to check rate limit status for
+    function isRateLimitEnabled(address tokenAddress) public view returns (bool) {
```
Comment (PraneshASP) [inline packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol:323]: nit: `rateLimitStatus` mapping is public. do you think we need this?

File: packages/solidity-contracts/test/behaviors/erc20GatewayV4.behavior.test.ts [hunk 4]
```diff
@@ -1340,6 +1353,10 @@ export function behavesLikeErc20GatewayV4(fixture: () => Promise<Env>) {
                 RATE_LIMIT_DURATION
               );
 
+            await erc20Gateway
+              .connect(deployer)
+              .updateRateLimitStatus(token, true);
+
```
Comment (PraneshASP) [inline packages/solidity-contracts/test/behaviors/erc20GatewayV4.behavior.test.ts:1359]: good to add a couple of tests to make sure the state is updated and the status update event is emitted.

File: packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol (hunk 3 above)
Comment (viraj124) [reply packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol:323]: just added for better readability that's the only reason

Comment (PraneshASP) [reply packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol:323]: I feel its redundant, but no harm in having it as well. feel free to keep it if you prefer.

Comment (viraj124) [reply packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol]: removed

**Review by PraneshASP — APPROVED** [review]

**Review by DefiCake — APPROVED** [review]

**Pull request merged**